pkg/types/ccipocr3/config.go

```diff
@@ -78,3 +64,148 @@ type ObserverInfo struct {
 	// Reads define the chains that the current node can read from.
 	Reads []ChainSelector `json:"reads"`
 }
+
+type P2PToSupportedChains map[P2PID]SupportedChains
+
+func (cm *P2PToSupportedChains) UnmarshalJSON(data []byte) error {
+	temp := make(map[string]SupportedChains)
+	if err := json.Unmarshal(data, &temp); err != nil {
+		return err
+	}
+	//if temp == nil {
+	//	*cm = nil
+	//	return nil
+	//}
+	*cm = make(P2PToSupportedChains)
+	for k, v := range temp {
+		var key P2PID
+		decoded, err := hex.DecodeString(k)
+		if err != nil {
+			return err
+		}
+		copy(key[:], decoded)
+		(*cm)[key] = v
+	}
+	return nil
+}
+
+func (cm P2PToSupportedChains) MarshalJSON() ([]byte, error) {
+	temp := make(map[string]SupportedChains)
+	for k, v := range cm {
+		temp[hex.EncodeToString(k[:])] = v
+	}
+	return json.Marshal(temp)
+}
+
+type ConsensusObservation struct {
+	// FChain defines the FChain value for each chain. FChain is used while forming consensus based on the observations.
+	FChain map[ChainSelector]int `json:"fChain"`
+	// PricedTokens is a list of tokens that we want to submit price updates for.
+	PricedTokens []types.Account `json:"pricedTokens"`
+	// NodeSupportedChains is a map of oracle IDs to SupportedChains.
+	NodeSupportedChains P2PToSupportedChains `json:"nodeSupportedChains"`
+}
+
+func (c ConsensusObservation) Validate() error {
+	for _, inf := range c.NodeSupportedChains {
+		for ch := range inf.Supported.Iter() {
+			if _, ok := c.FChain[ch]; !ok {
+				return fmt.Errorf("fChain not set for chain %d", ch)
+			}
+		}
+	}
+	if len(c.PricedTokens) == 0 {
+		return fmt.Errorf("priced tokens not set, at least one priced token is required")
+	}
+
+	return nil
+}
+
+type HomeChainConfig struct {
```

Comment:
Lets add a comment to this and other structures to make it clear what they represent and what they're used for